### DISCUSSION TITLE
How to configure to read MIDI from file (or local byte array)

### DISCUSSION THREAD

**Discussion by Gorillarock** [FortySevenEffects/arduino_midi_library#348] [General]
Basically the title. I have a midi file saved to an SD card. I read it into the Arduino and have a byte array.
Can I configure this lib to read from the byte array?
Or directly from the file?

**Comment by franky47** [comment]
Have a look at the [unit tests](https://github.com/FortySevenEffects/arduino_midi_library/blob/master/test/unit-tests/tests/unit-tests_MidiInput.cpp#L16-L18), there are examples on how to read from a buffer (array of bytes with an API that conforms to the Arduino Serial API on top).
